draft-homburg-deleg-incremental-deleg.md

```diff
@@ -0,0 +1,607 @@
+
+;; ADDITIONAL SECTION:
+ns.customer5.example.   3600    IN      A       198.51.100.5
+ns.customer5.example.   3600    IN      AAAA    2001:db8:5::1
+
+;; Query time: 0 msec
+;; EDNS: version 0; flags: do ; udp: 1232
+;; SERVER: 192.0.2.53
+;; WHEN: Tue Jul  2 10:55:07 2024
+;; MSG SIZE  rcvd: 593
+~~~
+{: #alias-response title="Aliasing referral response"}
+
+The incremental delegation of `customer7.example.` is alias to the one that is also used by `customer5.example.`
+Since both delegations are in the same zone, the authoritative name server for `example.` returns both the CNAME realising the alias, as well as the SVCB RRset which is the target of the alias in {{alias-response}}.
+In other cases an returned CNAME may need chasing by the resolver.
+{{Section 10.2 of !RFC9460}} states that zone structures that require following more than eight aliases (counting both AliasMode and CNAME records) are NOT RECOMMENDED.
+
+The resolver SHOULD register that the authoritative name server, supports incremental deleg for the duration indicated by the TTL for the CNAME, adjusted to the resolver's TTL boundaries, but only if it is longer than any already registered duration.
+
+### Referral responses with unsigned zones {#unsigned-zones}
+
+With unsigned zones, only if an incremental deleg delegation exists, the SVCB RRset (or CNAME) will be present in the authority section of referral responses.
+Resolvers SHOULD register that authoritative name server supports incremental deleg for the duration indicated by the TTL for the CNAME or SVCB RRset, adjusted to the resolver's TTL boundaries, but only if it is longer than any already registered duration.
+If the incremental deleg does not exist, then it is simply absent from the authority section and the referral response is indistinguishable from an non supportive authoritative.
+
+### Resolver behavior with authoritative name server support {#behavior-with-auth-support}
+
+Resolvers can detect support for incremental deleg with authoritative name servers from the returned referral responses.
+With DNSSEC signed zones, support is apparent with all referral responses, with unsigned zones only from referral responses for which a incremental delegation exists.
+For as long as the resolver knows that the authoritative name server support incremental deleg, no additional parallel direct deleg query (as described in {{recursive-resolver-behavior}}) needs to be send.
+
+However, if the resolver knows that the authoritative name server supports incremental deleg, *and* a DNSSEC signed zone is being served, then all referrals MUST contain either an incremental delegation, or NSEC(3) records showing that the delegation does not exist.
+If a referral is returned that does not contain an incremental delegation nor an indication that is does not exist, and , then the resolver MUST send an additional incremental deleg query to find the incremental delegation (or denial of its existence).
```

Comment:
s/is/it/
Not sure what comes after “and”

Reply:
Thanks, when I wrote it I was thinking: "and a _deleg is known to be present in the signed zone", but this is not necessary, because if it wouldn't have been there, then authoritative name server support would also not have been detected. Anyways, resolved in [08385ad](https://github.com/NLnetLabs/incremental-deleg/pull/2/commits/08385adfba80b3512da4d8132cdff551b17f6502)